TestMoveLeader is unstable

## Flaky Test

### Which jobs are failing
```
--- FAIL: TestMoveLeader (18.71s)
    member_test.go:334: move etcd leader does not return in 10 seconds
```

### CI link
https://github.com/tikv/pd/actions/runs/16284438006/job/45980322440?pr=9550

### Reason for failure (if possible)


### Anything else

--- a/tests/cluster.go
+++ b/tests/cluster.go
@@ -22,7 +22,6 @@ import (
 	"time"
 
 	"github.com/coreos/go-semver/semver"
-	"github.com/stretchr/testify/require"
 	clientv3 "go.etcd.io/etcd/client/v3"
 
 	"github.com/pingcap/errors"
@@ -42,7 +41,6 @@ import (
 	"github.com/tikv/pd/pkg/utils/keypath"
 	"github.com/tikv/pd/pkg/utils/logutil"
 	"github.com/tikv/pd/pkg/utils/syncutil"
-	"github.com/tikv/pd/pkg/utils/testutil"
 	"github.com/tikv/pd/server"
 	"github.com/tikv/pd/server/api"
 	"github.com/tikv/pd/server/apiv2"
@@ -170,13 +168,25 @@ func (s *TestServer) ResignLeader() error {
 }
 
 // ResignLeaderWithRetry resigns the leader of the server with retry.
-func (s *TestServer) ResignLeaderWithRetry(re *require.Assertions) {
+func (s *TestServer) ResignLeaderWithRetry() (err error) {
 	if !s.IsLeader() {
 		return
 	}
-	testutil.Eventually(re, func() bool {
-		return s.ResignLeader() == nil
-	})
+	// The default timeout of moving an etcd leader is 5 seconds,
+	// set the retry times to 3 will get a maximum of ~15 seconds of trying.
+	const retryCount = 3
+	for retry := range retryCount {
+		err = s.ResignLeader()
+		if err == nil {
+			return
+		}
+		// Do not retry if the last attempt fails.
+		if retry == retryCount-1 {
+			return
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	return
 }
 
 // State returns the current TestServer's state.

--- a/tests/integrations/tso/client_test.go
+++ b/tests/integrations/tso/client_test.go
@@ -517,7 +517,8 @@ func TestTSONotLeaderWhenRebaseErr(t *testing.T) {
 	re.NoError(failpoint.Enable("github.com/tikv/pd/server/rebaseErr", "return(true)"))
 	re.NoError(failpoint.Enable("github.com/tikv/pd/client/skipRetry", "return(true)"))
 	// Resign the leader to trigger the rebase error.
-	pdLeader.ResignLeaderWithRetry(re)
+	err = pdLeader.ResignLeaderWithRetry()
+	re.NoError(err)
 	// Trying to get TSO should fail with "not leader" error.
 	_, _, err = pdClient.GetTS(ctx)
 	re.ErrorContains(err, "not leader")
@@ -569,7 +570,8 @@ func TestMixedTSODeployment(t *testing.T) {
 	for range 2 {
 		n := r.Intn(2) + 1
 		time.Sleep(time.Duration(n) * time.Second)
-		leaderServer.ResignLeaderWithRetry(re)
+		err = leaderServer.ResignLeaderWithRetry()
+		re.NoError(err)
 		leaderServer = cluster.GetServer(cluster.WaitLeader())
 		re.NotNil(leaderServer)
 	}
@@ -708,7 +710,8 @@ func TestRetryGetTSNotLeader(t *testing.T) {
 
 	for range 5 {
 		time.Sleep(time.Second)
-		pdLeader.ResignLeaderWithRetry(re)
+		err = pdLeader.ResignLeaderWithRetry()
+		re.NoError(err)
 		leaderName = pdCluster.WaitLeader()
 		re.NotEmpty(leaderName)
 		pdLeader = pdCluster.GetServer(leaderName)

--- a/tests/server/member/member_test.go
+++ b/tests/server/member/member_test.go
@@ -302,7 +302,8 @@ func TestMoveLeader(t *testing.T) {
 	re.NotNil(originalLeaderServer)
 
 	// First, resign the original leader.
-	originalLeaderServer.ResignLeaderWithRetry(re)
+	err = originalLeaderServer.ResignLeaderWithRetry()
+	re.NoError(err)
 	newLeader := cluster.WaitLeader()
 	re.NotEmpty(newLeader)
 	re.NotEqual(originalLeader, newLeader)

--- a/tests/server/tso/tso_test.go
+++ b/tests/server/tso/tso_test.go
@@ -117,7 +117,8 @@ func TestDelaySyncTimestamp(t *testing.T) {
 	re.NoError(failpoint.Enable("github.com/tikv/pd/pkg/tso/delaySyncTimestamp", `return(true)`))
 
 	// Make the old leader resign and wait for the new leader to get a lease
-	leaderServer.ResignLeaderWithRetry(re)
+	err = leaderServer.ResignLeaderWithRetry()
+	re.NoError(err)
 	re.True(nextLeaderServer.WaitLeader())
 
 	ctx = grpcutil.BuildForwardContext(ctx, nextLeaderServer.GetAddr())

